### PR TITLE
add deployment id to CI/CD scripts

### PR DESCRIPTION
**Pull request by sunkickr**
add deployment id to docs to CI/CD docs to avoid encouraging users into a situation where they may accidentally deploy to the wrong deployment. Additionally specifying the deployment id make it so users avoid the issue outlined here: https://github.com/astronomer/docs/issues/2769

@jwitz should we add copy to warn users that if they don't specify a deployment id that run the risk of deploying to the wrong deployment and/or what is specified in the issue mentioned above?

### CODE REVIEW

File: astro/ci-cd-templates/github-actions.md [hunk 1]
```diff
@@ -70,6 +70,7 @@ To automate code deploys to a single Deployment using [GitHub Actions](https://g
         steps:
         - name: Deploy to Astro
           uses: astronomer/deploy-action@v0.2
+          deployment-id: <deployment-id>
```
Comment (jwitz) [inline astro/ci-cd-templates/github-actions.md:73]:
```suggestion
          deployment-id: <your-deployment-id>
```

File: astro/ci-cd-templates/github-actions.md [hunk 2]
```diff
@@ -169,6 +170,7 @@ If your Astro project requires additional build-time arguments to build an image
         - name: Build image
           uses: docker/build-push-action@v2
           with:
+            deployment-id: <deployment-id>
```
Comment (jwitz) [inline astro/ci-cd-templates/github-actions.md:173]:
```suggestion
            deployment-id: <your-deployment-id>
```

File: astro/ci-cd-templates/github-actions.md [hunk 3]
```diff
@@ -181,6 +183,7 @@ If your Astro project requires additional build-time arguments to build an image
         - name: Deploy to Astro
           uses: astronomer/deploy-action@v0.2
           with:
+            deployment-id: <deployment-id>
```
Comment (jwitz) [inline astro/ci-cd-templates/github-actions.md:186]:
```suggestion
            deployment-id: <your-deployment-id>
```

File: astro/ci-cd-templates/github-actions.md [hunk 4]
```diff
@@ -222,6 +225,7 @@ If your Astro project requires additional build-time arguments to build an image
...
```
Comment (jwitz) [inline astro/ci-cd-templates/github-actions.md:228]:
```suggestion
            deployment-id: <your-deployment-id>
```

**Review by jwitz — APPROVED** [review]

**Review by jwitz — COMMENTED** [review]
This LGTM, as long as our teams are consistently recommending these templates then I think we don't have to include an extra warning on top of improving the template. WDYT @erdos2n ? Would this have been enough to cover your edge case, or are folks still likely to build their own templates without the docs and miss this?

**Comment by erdos2n** [comment]
> 

Hey @jwitz , sorry I'm just seeing this. This will cover most of the edge case, but having some sort of documentation around self-hosted runners is necessary. For now, at least a warning that this process only applies when the deployment process isn't generating a persisted `config.yaml` file.

**Comment by lzdanski** [comment]
@jwitz - do you want to add the additional CI/CD warning to this PR, or would you like to assign the issue to me and I can add it so we can move to merge this PR?

**Comment by jwitz** [comment]
@erdos2n Sorry should have gotten your review on this before we merged, but we added a new note about self-hosted runners to the prerequisites! 
Screenshot 2023-09-20 at 9 47 37 AM

**Pull request merged**